https://bugs.gentoo.org/450714
http://sourceware.org/bugzilla/show_bug.cgi?id=15007

From 4220c3ef773c0e7fff7c95ecd7cff69886fe4c0e Mon Sep 17 00:00:00 2001
From: Andreas Jaeger <aj@suse.de>
Date: Sat, 27 Apr 2013 16:17:55 +0200
Subject: [PATCH] Fix guards for qecvt

	[BZ #15007]
	* stdlib/stdlib.h: Update guards for qecvt.
	* stdlib/bits/stdlib-ldbl.h: Sync guards for qecvt etc with
	<stdlib.h>.
---
 ChangeLog                 | 7 +++++++
 NEWS                      | 8 ++++----
 stdlib/bits/stdlib-ldbl.h | 4 +---
 stdlib/stdlib.h           | 7 +++----
 4 files changed, 15 insertions(+), 11 deletions(-)

diff --git a/stdlib/bits/stdlib-ldbl.h b/stdlib/bits/stdlib-ldbl.h
index 1e1bcef..a606fd3 100644
--- a/stdlib/bits/stdlib-ldbl.h
+++ b/stdlib/bits/stdlib-ldbl.h
@@ -30,12 +30,10 @@ __END_NAMESPACE_C99
 __LDBL_REDIR1_DECL (strtold_l, strtod_l)
 #endif
 
-#if defined __USE_SVID || defined __USE_XOPEN_EXTENDED
-# ifdef __USE_MISC
+#ifdef __USE_MISC
 __LDBL_REDIR1_DECL (qecvt, ecvt)
 __LDBL_REDIR1_DECL (qfcvt, fcvt)
 __LDBL_REDIR1_DECL (qgcvt, gcvt)
 __LDBL_REDIR1_DECL (qecvt_r, ecvt_r)
 __LDBL_REDIR1_DECL (qfcvt_r, fcvt_r)
-# endif
 #endif
diff --git a/stdlib/stdlib.h b/stdlib/stdlib.h
index 4b83944..99a830d 100644
--- a/stdlib/stdlib.h
+++ b/stdlib/stdlib.h
@@ -823,9 +823,9 @@ extern char *fcvt (double __value, int __ndigit, int *__restrict __decpt,
    be written to BUF.  */
 extern char *gcvt (double __value, int __ndigit, char *__buf)
      __THROW __nonnull ((3)) __wur;
+#endif
 
-
-# ifdef __USE_MISC
+#ifdef __USE_MISC
 /* Long double versions of above functions.  */
 extern char *qecvt (long double __value, int __ndigit,
 		    int *__restrict __decpt, int *__restrict __sign)
@@ -854,8 +854,7 @@ extern int qfcvt_r (long double __value, int __ndigit,
 		    int *__restrict __decpt, int *__restrict __sign,
 		    char *__restrict __buf, size_t __len)
      __THROW __nonnull ((3, 4, 5));
-# endif	/* misc */
-#endif	/* use MISC || use X/Open Unix */
+#endif	/* misc */
 
 
 __BEGIN_NAMESPACE_STD
-- 
1.8.2.1

